lib/src/rules/prefer_single_quotes.dart

```diff
@@ -117,7 +122,8 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitSimpleStringLiteral(SimpleStringLiteral string) {
-    if (string.isSingleQuoted || string.value.contains("'")) {
+    if (useSingle && (string.isSingleQuoted || string.value.contains("'")) ||
```

Comment:
Perhaps extract a function that can be used here and below?

Reply:
The conditions are not exactly the same.